### PR TITLE
175v2 add consep segmentation dataset

### PR DESCRIPTION
**Pull request by roman807**
closes https://github.com/kaiko-ai/eva/issues/175

(https://github.com/kaiko-ai/eva/pull/517 should be merged first)

### CODE REVIEW

File: src/eva/vision/data/datasets/segmentation/consep.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        image_path = self._get_image_path(sample_index)
+        image_shape = io.fetch_nifti_shape(image_path)
+        return image_shape[-1]
+
+    def _fetch_samples_names(self) -> List[str]:
+        """Returns the name of all the samples of the selected dataset split."""
+
+        def _samples_dirs_split(split: str) -> List[str]:
+            split_dir = os.path.join(self._root, f"{split}/Images")
+            return [
```
Comment (nkaenzig) [inline src/eva/vision/data/datasets/segmentation/consep.py:139]: Nit: All images seem to be `.png`, so using glob might be safer / more concise:
something like 
```python
[os.path.basename(f) for f in glob.glob(os.path.join(self._root, f"{split_dir}/*.png"))]
```

File: src/eva/vision/data/datasets/segmentation/consep.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        return self.get_file_path(sample_index, "Images", ".png")
+
+    def _get_mask_path(self, sample_index: int) -> str:
+        """Returns the directory of the corresponding masks."""
+        return self.get_file_path(sample_index, "Labels", ".mat")
+
+    def _get_number_of_slices_per_sample(self, sample_index: int) -> int:
+        """Returns the total amount of slices of a sample."""
+        image_path = self._get_image_path(sample_index)
+        image_shape = io.fetch_nifti_shape(image_path)
```
Comment (nkaenzig) [inline src/eva/vision/data/datasets/segmentation/consep.py:131]: does that belong to consep? is there nifti data?

**Review by nkaenzig — COMMENTED** [review]
Looks good, left some minor comments.

File: src/eva/vision/data/datasets/segmentation/consep.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    @override
+    def filename(self, index: int) -> str:
+        return self._samples_names[index]
+
+    @override
+    def prepare_data(self) -> None:
+        _validators.check_dataset_exists(self._root, True)
+
+    @override
+    def configure(self) -> None:
```
Comment (nkaenzig) [inline src/eva/vision/data/datasets/segmentation/consep.py:77]: I think the `configure` method is not meant to be overwritten, only `setup`, `prepare_data` and `validate`

Comment (nkaenzig) [reply src/eva/vision/data/datasets/segmentation/consep.py:77]: ups, nevermind, I was mistaken, `setup` seems to be the one we shouldn't overwrite

**Comment by roman807** [comment]
will be covered here: https://github.com/kaiko-ai/eva/pull/547

**Pull request closed without merging**